core/evm.go

```diff
@@ -64,18 +64,19 @@ func NewEVMBlockContext(header *types.Header, chain ChainContext, author *common
 		random = &header.MixDigest
 	}
 	return vm.BlockContext{
-		CanTransfer: CanTransfer,
-		Transfer:    Transfer,
-		GetHash:     GetHashFn(header, chain),
-		Coinbase:    beneficiary,
-		BlockNumber: new(big.Int).Set(header.Number),
-		Time:        header.Time,
-		Difficulty:  new(big.Int).Set(header.Difficulty),
-		BaseFee:     baseFee,
-		BlobBaseFee: blobBaseFee,
-		GasLimit:    header.GasLimit,
-		Random:      random,
-		L1CostFunc:  types.NewL1CostFunc(config, statedb),
+		CanTransfer:      CanTransfer,
+		Transfer:         Transfer,
+		GetHash:          GetHashFn(header, chain),
+		Coinbase:         beneficiary,
+		BlockNumber:      new(big.Int).Set(header.Number),
+		Time:             header.Time,
+		Difficulty:       new(big.Int).Set(header.Difficulty),
+		BaseFee:          baseFee,
+		BlobBaseFee:      blobBaseFee,
+		GasLimit:         header.GasLimit,
+		Random:           random,
+		L1CostFunc:       types.NewL1CostFunc(config, statedb),
+		OperatorCostFunc: types.NewOperatorCostFunc(config, statedb),
```

Comment:
I lean towards not adding a new hook and just making the L1 cost function more generic, a better name for it would be `L2CostFunc` which would imply it does the L2 specific fee calculations. Curious what other people think

Reply:
I thought about this too. Currently, these are the function signatures for the L1CostFunc and the OperatorCostFunc

```
type L1CostFunc func(rcd RollupCostData, blockTime uint64) *big.Int

type OperatorCostFunc func(gasUsed *big.Int, isRefund bool, blockTime uint64) *big.Int

```

I could potentially package up the `gasUsed` and `isRefund` together into an `OperatorCostData` or something. Then, we'd have this type.

```
type L2CostFunc func(rcd RollupCostData, ocd OperatorCostData, blockTime uint64) (*big.Int, *big.Int)
```
which returns the l1 cost and operator cost as a pair.

This pattern seems clunkier to me -- though the L1 fee and Operator fee are conceptually similar, I don't like returning both in a single function. For example, when calculating refunds, we only need the operator fee but not the L1 fee. 

I'd definitely be open to unifying the two with a cleaner function design though.